### PR TITLE
Upgrade some dependencies

### PR DESCRIPTION
**Pull request by eranrund**
Hi!

We have a project that links newer version of `curve25519-dalek`, `sha3` and `digest`. This PR updates those. I haven't updated the version in `Cargo.toml` since I am not sure how you'd like to go about this if you choose to accept this PR.
Unfortunately, due to a dependency on `clear_on_drop` which now uses `llvm_asm!`, the Rust toolchain version had to be upgraded.

Thanks and please let me know if there's anything I can help with to get this merged in!

### CODE REVIEW

**Comment by eranrund** [comment]
Ping? Any chance in getting this merged?

**Comment by hdevalence** [comment]
Hey, sorry for the delay.  I missed a bunch of stuff in August because I had to evacuate from a wildfire.

File: src/inner_product_proof.rs [hunk 1]
```diff
@@ -338,6 +338,7 @@ impl InnerProductProof {
     /// The layout of the inner product proof is:
     /// * \\(n\\) pairs of compressed Ristretto points \\(L_0, R_0 \dots, L_{n-1}, R_{n-1}\\),
     /// * two scalars \\(a, b\\).
+    #[allow(dead_code)]
```
Comment (hdevalence) [inline src/inner_product_proof.rs:341]: Hmm, just to clarify, this is suppressing an existing warning, not a new warning that appeared as a result of these changes?

**Comment by eranrund** [comment]
> Hey, sorry for the delay. I missed a bunch of stuff in August because I had to evacuate from a wildfire.

Very sorry to hear you were affected by the fires, and I hope you and your close ones are doing well.

File: src/inner_product_proof.rs (hunk 1 above)
Comment (eranrund) [reply src/inner_product_proof.rs:341]: Correct

**Comment by dunnock** [comment]
Was about to submit similar request but then found this one is pending, any chance to merge this one in? Thanks!

**Comment by isislovecruft** [comment]
Hi @eranrund! Thanks for the PR.

It looks like similar changes to these were merged [here](https://github.com/dalek-cryptography/bulletproofs/pull/334/files). If you still need the warning suppression and the compiler upgraded, could you please make separate PRs for those issues?

**Comment by eranrund** [comment]
Thank you @isislovecruft, I think this is good enough.

**Pull request closed without merging**